pkg/providers/tinkerbell/template.go

```diff
@@ -640,10 +638,11 @@ func generateTemplateBuilder(clusterSpec *cluster.Spec) (providers.TemplateBuild
 }
 
 // GenerateNoProxyList generates NOPROXY list for tinkerbell provider based on HTTP_PROXY, HTTPS_PROXY, NOPROXY and tinkerbellIP.
-func GenerateNoProxyList(clusterSpec *v1alpha1.Cluster, datacenterSpec v1alpha1.TinkerbellDatacenterConfigSpec, tinkerbellIP string) []string {
+func GenerateNoProxyList(clusterSpec *v1alpha1.Cluster, datacenterSpec v1alpha1.TinkerbellDatacenterConfigSpec) []string {
```

Comment:
This is a local function that's not used outside this package. Let's un-export it
```suggestion
func generateNoProxyList(clusterSpec *v1alpha1.Cluster, datacenterSpec v1alpha1.TinkerbellDatacenterConfigSpec) []string {
```